deploy/helm/templates/deployment.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "starboard-operator.fullname" . }}
+  labels:
+    {{- include "starboard-operator.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "starboard-operator.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- . | toYaml | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "starboard-operator.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: "{{ .Chart.Name }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          env:
+            - name: OPERATOR_NAMESPACE
+              value: {{ .Release.Namespace | quote }}
+            - OPERATOR_TARGET_NAMESPACES:
+              value: {{ .Values.targetNamespaces | quote }}
+            - name: OPERATOR_METRICS_BIND_ADDRESS
+              value: ":8080"
+            - name: OPERATOR_HEALTH_PROBE_BIND_ADDRESS
+              value: ":9090"
```

Comment:
I locked in some configuration which should not be changed by the user, and k8s api-server will error if someone tries to add such configuration manually.

It was important to have targetNamespace easily accessible, no matter if the user is referencing an external k8s secret as well, because that is the information used to deduce if we should add ClusterRole resources or just Role resources for example. ClusterRole is good to avoid if we only need a namespaced version (Role).